core/js/src/main/scala/bobcats/HashPlatform.scala

```diff
@@ -16,14 +16,31 @@
 
 package bobcats
 
-import cats.effect.kernel.Async
+import cats.effect.kernel.{Async, Resource, Sync}
 import cats.syntax.all._
 import scodec.bits.ByteVector
 
+private final class NodeCryptoDigest[F[_]](var hash: facade.node.Hash, algorithm: String)(
+    implicit F: Sync[F])
+    extends UnsealedDigest[F] {
+  override def update(data: ByteVector): F[Unit] = F.delay(hash.update(data.toUint8Array))
+  override val reset = F.delay {
+    hash = facade.node.crypto.createHash(algorithm)
+  }
+  override def get: F[ByteVector] = F.delay(ByteVector.view(hash.digest()))
+}
+
 private[bobcats] trait HashCompanionPlatform {
   implicit def forAsync[F[_]](implicit F: Async[F]): Hash[F] =
     if (facade.isNodeJSRuntime)
       new UnsealedHash[F] {
+
+        override def incremental(algorithm: HashAlgorithm): Resource[F, Digest[F]] =
+          Resource.make(F.catchNonFatal {
+            val alg = algorithm.toStringNodeJS
+            val hash = facade.node.crypto.createHash(alg)
```

Comment:
Isn't this side-effecting? e.g. should be a `delay`

Reply:
Thanks! I opened this up to think on it. I'm not very happy with it yet.

I have a question; the current code uses `MessageDigest#getInstance` within a `catchNonFatal`, but technically this is impure; should we wrap this in a `delay`? It's fairly common to register providers after initialization.

Reply:
Hmm yeah. When in doubt err on side of `delay` I suppose and

> It's fairly common to register providers after initialization.

this sounds like a legitimate reason to use `delay`.

Reply:
Thinking about it, it is probably best to be conservative for now.

I suspect that certainly for downstream library maintainers, they may need to have access to certain algorithms which aren't available in the Standard JDK and getting people to deal with those config files is probably a no-go.